pkg/manager/impl/node_execution_manager.go

```diff
@@ -225,6 +226,19 @@ func (m *NodeExecutionManager) CreateNodeEvent(ctx context.Context, request admi
 	}
 	m.dbEventWriter.Write(request)
 
+	// If the request references a dynamically compiled subworkflow in the node metadata, serialize that to the db.
+	if request.Event.GetTaskNodeMetadata() != nil && request.Event.GetTaskNodeMetadata().DynamicWorkflow != nil {
+		_, err := util.WriteCompiledWorkflow(ctx, m.db, m.storagePrefix, m.storageClient,
```

Comment:
The prefix here is not specific to this execution, right? I think the path we store this workflow in should be sandboxed to the execution (e.g. `s3://myprefix/admin/project/domain/name/node1/sub-node/sub-node/<trial-num>/workflow_id`) because the same id can be returned multiple times (it's dynamically generated after all and the shape of the graph can depend on the inputs to the generator task)...
do you agree?

you don't have to have this path... just a suggestion... you can also hash the string and just keep it as a prefix of the workflow id (thought maybe maintaining the exec hierarchy is useful for debugging? what do you think?

Reply:
oh good point - yes a human readable execution id in the closure path sounds like a good and readable guarantee of uniqueness! ty

Reply:
> The prefix here is not specific to this execution
No, but we do append the workflow keys to the prefix path to get the full closure loc

Reply:
Do we register it? or just store it in storage?

Reply:
I think we might actually change that rand to be the hash of the workflow closure for consistency...

Reply:
yes, why even worry about de-duping in S3? Does it really matter? We can simply write new keys right? And cache the key. The semantics are that, every write to workflow closure has a full copy of the closure, so does not matter, which one we consider as a copy?
Does the determinism help with anything?